netlogo-gui/src/main/lab/Worker.scala

```diff
@@ -107,8 +107,38 @@ class Worker(val protocol: LabProtocol)
         val world = ws.world
         var d = world.getDimensions
         for((name, value) <- settings) {
-          if(world.isDimensionVariable(name)) {
             val v = value.asInstanceOf[java.lang.Double].intValue
```

Comment:
Values can also be `java.lang.Boolean` and/or `String`. Even dimension variables may be incorrectly set to a string. If you want, this can be thoroughly handled by pattern-matching. Alternatively, this can go inside the check for a dimension variable.